docs/developing.rst

```diff
@@ -373,6 +373,64 @@ workflow, with jobs defined within a job or job function being created at
 run time.
 
 
+Developing pipelines with toil-lib
+----------------------------------
+
+`toil-lib`_ provides many functions and job functions that are useful in
+developing pipelines in Toil. Installation instructions for toil-lib are
+available `here`_.
+
+.. _toil-lib: https://github.com/BD2KGenomics/toil-lib/
+
+.. _here: https://github.com/BD2KGenomics/toil-lib/blob/master/docs/installation.rst
+
+toil-lib packages and functions can be imported to a Toil pipeline like any
+Python package. Jobs from toil-lib can then be added to workflows staticly.
+For example::
+
+    from toil.job import Job
+    from toil_lib.urls import download_url
+
+    def helloWorld(job, message, memory="2G", cores=2, disk="3G"):
+        job.fileStore.logToMaster("Hello world, "
+        "I have a message: %s" % message) # This uses a logging function
+        # of the Job.FileStore class
+
+    j1 = Job.wrapJobFn(download_url, "https://sample.url")
+    j2 = j1.addChildJobFn(helloWorld, "second or third")
+    j3 = j1.addChildJobFn(helloWorld, "second or third")
+    j4 = j2.addChildJobFn(helloWorld, "last")
+    j3.addChild(j4)
+
+    if __name__=="__main__":
+        options = Job.Runner.getDefaultOptions("./toilWorkflowRun")
+        options.logLevel = "INFO"
+        Job.Runner.startToil(j1, options)
+
+
+Using Docker containers in Toil
+-------------------------------
+
+Docker containers are commonly used with Toil to maintain the portability of
+workflows. In order to use Docker containers with Toil, Docker must be installed
+on all workers of a cluster in order to run a workflow containing Docker containers.
+Instructions for installing Docker can be found on the `Docker`_ website.
```

Comment:
I would also link to some documentation about setting up Docker so it can be run without sudo, since this is an assumption that we make in `docker_call`.